core/sdk-core/src/test/java/software/amazon/awssdk/core/retry/RetryPolicyTest.java

```diff
@@ -185,4 +186,16 @@ public void fastFailRateLimitingConfigured_retryModeAdaptive_doesNotThrow() {
     public void hashCodeDoesNotThrow() {
         RetryPolicy.defaultRetryPolicy().hashCode();
     }
+
+    @Test
+    public void fullJitter_zeroBackoffTimeZeroDelayZeroRetries_shouldNotThrowIllegalArgumentException() {
+        RetryPolicyContext context = RetryPolicyContext.builder()
+                                                       .retriesAttempted(0)
+                                                       .build();
+        FullJitterBackoffStrategy strategy = FullJitterBackoffStrategy.builder()
+                                                            .baseDelay(Duration.ZERO)
+                                                            .maxBackoffTime(Duration.ZERO)
+                                                            .build();
+        assertDoesNotThrow(() -> strategy.computeDelayBeforeNextRetry(context));
```

Comment:
Should we also validate the output is correct?

Reply:
Done. I removed `assertDoesNotThrow` because strategy.computeDelayBeforeNextRetry will be called to test the output anyway.